code/modules/clothing/shoes/bananashoes.dm

```diff
@@ -37,6 +37,22 @@
 	else
 		to_chat(user, "<span class='warning'>You cannot retrieve any bananium from the prototype shoes!</span>")
 
+/obj/item/clothing/shoes/clown_shoes/banana_shoes/attackby(obj/item/O, mob/user, params)
+	var/datum/component/material_container/bananium = GetComponent(/datum/component/material_container)
```

Comment:
Avoid using GetComponent and send some signals instead, seems like `COMSIG_PARENT_ATTACKYBY` would be what you're looking for here.
https://tgstation13.org/phpBB/viewtopic.php?f=5&t=22674

Reply:
this is the code that I was restoring from the deleted edit, I'm not sure how you would use signals to do this

Reply:
The shoes work fine. The material container component already listens to COMSIG_PARENT_ATTACKYBY.
If I should take a guess, the two issues were submitted by people who tried to add bananium ore instead of sheets.